### PR TITLE
Save prediction results

### PR DESCRIPTION
**Pull request by matkrasinski**
-  ## Done saving generated objects from AI module to database with the possibility of displaying and comparing them in QGIS
## Example:
Zrzut ekranu 2023-06-16 o 13 43 17

### CODE REVIEW

File: src/main/java/pl/ioad/adoto/backend/ai/logic/service/AiService.java [hunk 1]
```diff
@@ -15,9 +17,19 @@
...
 
     private final AICommunicationFacade aiCommunicationFacade;
+    private final DBService dbService;
     private final Map<String, Layer> layers;
 
     public List<List<AiResult>> getAiResults(AiRequest aiRequest) {
-        return aiCommunicationFacade.getAiResults(aiRequest.width(), layers.get(aiRequest.layer()).getAiSpell(), aiRequest.base64Image());
+        var aiResults = aiCommunicationFacade.getAiResults(aiRequest.width(),
+                layers.get(aiRequest.layer()).getAiSpell(), aiRequest.base64Image());
+        if (aiResults != null && !aiResults.isEmpty())
```
Comment (RadekBucki) [inline src/main/java/pl/ioad/adoto/backend/ai/logic/service/AiService.java]: It's not good practise to miss brackets, it's not python

File: src/main/java/pl/ioad/adoto/database/DBService.java [hunk 2]
```diff
@@ -18,29 +21,102 @@
...
+                .toList();
+    }
+
+    public void savePrediction(List<Double> x,
+                                    List<Double> y,
+                                    Double minX,
+                                    Double minY,
+                                    Double maxX,
+                                    Double maxY,
+                                    String entitiesType) {
```
Comment (RadekBucki) [inline src/main/java/pl/ioad/adoto/database/DBService.java]: Please use EntitiesType enum instead if string

File: src/main/java/pl/ioad/adoto/database/DBService.java [hunk 3]
```diff
@@ -18,29 +21,102 @@
...
+        var xStep = (maxX - minX) / WINDOW_WIDTH;
+        var yStep = (maxY - minY) / WINDOW_WIDTH;
+
+        var geoX = x.stream().map(xCord -> minX + (xCord * xStep)).toList();
+        var geoY = y.stream().map(yCord -> minY + ((WINDOW_WIDTH - yCord) * yStep)).toList();
+
+        List<Coordinate> coordinates = new ArrayList<>();
+
+        for (int i = 0; i < geoX.size(); i++)
```
Comment (RadekBucki) [inline src/main/java/pl/ioad/adoto/database/DBService.java]: Same brackets

File: src/main/java/pl/ioad/adoto/database/DBService.java [hunk 4]
```diff
@@ -18,29 +21,102 @@
...
+            coordinates.add(new Coordinate(geoX.get(i), geoY.get(i)));
+
+
+        if (entitiesType.equals("BUILDING") || entitiesType.equals("FOREST"))
```
Comment (RadekBucki) [inline src/main/java/pl/ioad/adoto/database/DBService.java]: Brackets

File: src/main/java/pl/ioad/adoto/database/DBService.java (hunk 3 above)
Comment (RadekBucki) [reply src/main/java/pl/ioad/adoto/database/DBService.java]: also it will be good to use stream functionality instead of for ;)

**Review by RadekBucki — CHANGES_REQUESTED** [review]

**Review by RadekBucki — APPROVED** [review]

**Review by kambed — APPROVED** [review]

**Pull request merged**